src/System.Collections/src/System/Collections/Generic/EqualityComparer.cs

```diff
@@ -0,0 +1,336 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+/*============================================================
+**
+** Class: EqualityComparer<T> 
+**
+===========================================================*/
+
+using System;
+using System.Collections;
+
+namespace System.Collections.Generic
+{
+    public abstract class EqualityComparer<T> : IEqualityComparer, IEqualityComparer<T>
+    {
+        protected EqualityComparer()
+        {
+        }
+
+        public static EqualityComparer<T> Default
+        {
+            get
+            {
+                if (_default == null)
+                {
+                    object comparer;
+
+                    // NUTC compiler is able to static evalulate the conditions and only put the necessary branches in finally binary code, 
```

Comment:
EqualityComparer is type forwarded to mscorlib, I've updated my initial commit to remove this file.